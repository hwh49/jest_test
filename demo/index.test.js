import {sum, subtract} from './index'
require('../jsdom-config')
test('sum: ', () => {
    expect(sum(1,2)).toBe(3)
})

test('subtract: ', () => {
    expect(subtract(1,2)).toBe(-1)
})

function renderHtml() {
    const div = document.createElement("div");
    div.innerHTML = "<h1>Hello World</h1>";
    document.body.appendChild(div);
}

test("DOM 测试：", () => {
    renderHtml();
    expect(document.querySelector("h1").innerHTML).toBe("Hello World");
});

test("Snapshot Test:", () => {
    expect(document.body).toMatchSnapshot(); // 首次执行会生成快照文件。
});