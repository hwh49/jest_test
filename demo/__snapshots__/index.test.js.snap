// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test: 1`] = `
<body>
  <div>
    <h1>
      Hello World
    </h1>
  </div>
</body>
`;
